apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/Labels.cs

```diff
@@ -0,0 +1,127 @@
+        /// Creates a a map of labels to represent a <see cref="StackTrace"/> for a span.
+        /// </summary>
+        public static Dictionary<string, string> FromStackTrace(StackTrace stackTrace)
+        {
+            GaxPreconditions.CheckNotNull(stackTrace, nameof(stackTrace));
+            return new Dictionary<string, string>()
+            {
+                { StackTrace, GenerateJsonStringStackTrace(stackTrace) }
+            };
+        }
+
+        /// <summary>
+        /// Gets a map with the label for the agent which contains the agent's name and version.
+        /// </summary
+        internal static Dictionary<string, string> GetAgentLabel(System.Type type) =>
+            new Dictionary<string, string> { { Agent, $"{AgentName} {GetAgentVersion(type)}" } };
+
+        /// <summary>Gets the version of the current library using reflection.</summary>
+        internal static string GetAgentVersion(System.Type type)
+        {
+            return type.GetTypeInfo()
+                .Assembly
+                .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                .InformationalVersion;
+        }
+
+        /// <summary>
+        /// Creates a string JSON representation of a stack trace or the empty string
+        /// if the stack trace has no frames.
+        /// </summary>
+        private static string GenerateJsonStringStackTrace(StackTrace stackTrace)
+        {
+            if (stackTrace.GetFrames().Length == 0)
+            {
+                return string.Empty;
+            }
+
+            using (StringWriter sw = new StringWriter())
+            {
+                JsonWriter writer = new JsonTextWriter(sw);
+                writer.WriteStartObject();
+                writer.WritePropertyName("stack_frame");
+                writer.WriteStartArray();
+
+                foreach (StackFrame stackFrame in stackTrace.GetFrames())
+                {
+                    writer.WriteStartObject();
+                    writer.WritePropertyName("class_name");
+                    writer.WriteValue(stackFrame.GetMethod().DeclaringType.Name);
+                    writer.WritePropertyName("method_name");
+                    writer.WriteValue(stackFrame.GetMethod().Name);
+
+                    if (stackFrame.GetFileName() != null)
+                    {
+                        writer.WritePropertyName("file_name");
+                        writer.WriteValue(stackFrame.GetFileName());
+                    }
+
+                    if (stackFrame.GetFileLineNumber() != 0)
+                    {
+                        writer.WritePropertyName("line_number");
+                        writer.WriteValue(stackFrame.GetFileLineNumber());
+                    }
+                    writer.WriteEndObject();
+                }
+
+                writer.WriteEndArray();
+                writer.WriteEndObject();
+                writer.Close();
+
+                return sw.GetStringBuilder().ToString();
```

Comment:
Actually, `.GetStringBuilder()` is unnecessary: the `ToString` implementation of `StringWriter` returns the text of the underlying builder.